### PR TITLE
[NOJIRA] Fix table components to accept arbitrary props and custom cl…

### PR DESCRIPTION
**Pull request by matthewdavidson**
…ass names

+ [x] For any new components I've made sure that the style can be extended by the consumer using a `className` override.
+ [x] For any new files I've included the [Apache 2.0 License header](https://github.com/Skyscanner/backpack/blob/master/packages/bpk-tokens/formatters/license-header.js#L2-L16) at the top of the file.
+ [x] I've updated `changelog.md` for any changes that need to be highlighted in the changelog.
+ [x] If I've updated `npm-shrinkwrap.json` those changes are intentional.

### CODE REVIEW

File: packages/bpk-component-table/src/BpkTableHead.js [hunk 1]
```diff
@@ -19,7 +19,7 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 
-const BpkTableHead = props => <thead>{props.children}</thead>;
+const BpkTableHead = props => <thead {...props} />;
```
Comment (k0nserv) [inline packages/bpk-component-table/src/BpkTableHead.js]: Does passing `children` like this have any caveats or does it just compile to identical code?

Comment (k0nserv) [inline packages/bpk-component-table/src/BpkTableHead.js]: Just checked

```javascript
const test1 = <b>Hello</b>
const test2 = <b children={["hello"]} />
```
Compiles to

```javascript
"use strict";

var test1 = React.createElement(
  "b",
  null,
  "Hello"
);
var test2 = React.createElement("b", { children: ["hello"] });
```

Comment (k0nserv) [inline packages/bpk-component-table/src/BpkTableHead.js]: BUt that probably just takes the same code paths anyway

Comment (matthewdavidson) [inline packages/bpk-component-table/src/BpkTableHead.js]: We use this pattern elsewhere and _it works_... not sure of the implications.

Comment (k0nserv) [inline packages/bpk-component-table/src/BpkTableHead.js]: https://github.com/facebook/react/blob/master/src/isomorphic/classic/element/ReactElement.js#L214-L230 it gets added to props when passed in third argument

**Comment by georgegillams** [comment]
...except for the "failed build" thing...

**Pull request merged**